src/main/java/cam72cam/immersiverailroading/model/part/PartSound.java

```diff
@@ -29,6 +29,14 @@ public void effects(EntityMoveableRollingStock stock, boolean enabled) {
     public void effects(EntityMoveableRollingStock stock, float volume) {
         effects(stock, volume, 1);
     }
+    
+    public boolean isPlaying(EntityMoveableRollingStock stock){
+     	ISound sound = sounds.get(stock.getUUID());
+	if(sound == null){
+    		return false;
+    	}
+     	return sound.isPlaying();
```

Comment:
This works fine. I personally would have used a ternary expression, but that's my personal style.
```java
return sound != null && sound.isPlaying();
```